### DISCUSSION TITLE
FIRE optimization algorism need to be improved

### DISCUSSION THREAD

**Discussion by QuantumMisaka** [deepmodeling/abacus-develop#5269] [General]
### Details

As https://github.com/deepmodeling/abacus-develop/issues/3558, the BFGS is correct and have good efficiency now, but FIRE algorism is still lack of efficiency. 

ASE also have FIRE algorism, we can take it as a reference

### Task list for Issue attackers (only for developers)

- [ ] Reproduce the performance issue on a similar system or environment.
- [ ] Identify the specific section of the code causing the performance issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified performance issue.
- [ ] Implement the chosen solution to address the performance issue.
- [ ] Test the implemented solution to ensure it improves performance without introducing new issues.
- [ ] Optimize the solution if necessary, considering trade-offs between performance and other factors (e.g., code complexity, readability, maintainability).
- [ ] Review and incorporate any relevant feedback from users or developers.
- [ ] Merge the improved solution into the main codebase and notify the issue reporter.

**Comment by WHUweiqingzhou** [comment]
Thanks for your suggestion!
